### PR TITLE
docs(DST-420): add pattern section

### PR DESCRIPTION
**Pull request by sebald**
- create a new top level section for patterns
- move pattern guidelines to the section (multi selection and forms)

### CODE REVIEW

File: docs/ui/TeaserCard.tsx [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+          {icon}
+        </div>
+        <div className="text-2xl font-semibold tracking-tight">{title}</div>
+        <div className="text-text-primary-muted text-sm">{caption}</div>
+      </Stack>
+    </Card>
+  </Link>
+);
+
+export interface TeaserListProps {
```
Comment (aromko) [inline docs/ui/TeaserCard.tsx:27]: The file is atm small so it is fine but in general I would suggest to place all sort of type definitions at the top of file 😬  but maybe just my personal gusto

**Review by aromko — APPROVED** [review]

File: docs/ui/TeaserCard.tsx (hunk 1 above)
Comment (sebald) [reply docs/ui/TeaserCard.tsx:27]: I like to have it next to the component is used :D But yeah we can definitely agree on something and add it to our dev docs?

**Review by aromko — DISMISSED** [review]

**Comment by sarahgm** [comment]
@sebald conflicts

File: docs/ui/TeaserCard.tsx [hunk 2]
```diff
@@ -0,0 +1,37 @@
+import { Card, Stack, Tiles } from '@/ui';
+import type { ReactElement } from 'react';
+
+import Link from 'next/link';
+
+export interface TeaserCardProps {
```
[the remark this answers is not in the document]
Comment (sebald) [reply docs/ui/TeaserCard.tsx:6]: yeah, but future proofing it :D

**Review by sarahgm — APPROVED** [review]

**Pull request merged**